migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs

```diff
@@ -563,7 +566,13 @@ fn foreign_keys_match(previous: &ForeignKeyWalker<'_>, next: &ForeignKeyWalker<'
             .constrained_columns()
             .zip(next.constrained_columns())
             .all(|(previous, next)| {
-                previous.name() == next.name() && previous.column_type_family() == next.column_type_family()
+                let families_match = match (previous.column_type_family(), next.column_type_family()) {
+                    (ColumnTypeFamily::Uuid, ColumnTypeFamily::String) => true,
+                    (ColumnTypeFamily::String, ColumnTypeFamily::Uuid) => true,
+                    (x, y) => x == y,
```

Comment:
that works as a fix for now, but we have to find another mechanism here in general

Reply:
yep. I don't like this but at least now we have a test that helps when we remove the type families.